### PR TITLE
Initial config

### PR DESCRIPTION
**Pull request by JorikSchellekens**
Base project.

- Typescript
- Prettier, husky, and lint-staged to prettifying and lint on stage
- SCSS support
- eslint using matrix-js-sdk lint rules which are cleaned up a bit

fixes: https://github.com/matrix-org/matrix.to/issues/82

### CODE REVIEW

**Comment by t3chguy** [comment]
> eslint using matrix-js-sdk lint rules which are cleaned up a bit

Any reason we're not importing the js-sdk eslint rules like the other projects do?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "lint-staged": {
+    "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
```
Comment (t3chguy) [inline package.json:32]: might be nice to get opinions from other riot-web team members as I am not at all familiar with Husky

**Comment by JorikSchellekens** [comment]
> > eslint using matrix-js-sdk lint rules which are cleaned up a bit
> 
> Any reason we're not importing the js-sdk eslint rules like the other projects do?


The js sdk's eslintrc explicitly relaxes some of its ideals as a result of some debt in the js sdk. The majority of the rules come from eslint:recommended and google. I think it's cleaner to have a more minimal slate than import the js sdk's and then tighten up our configuration again.

Potentially controversial opinion: The js-sdk's eslint shouldn't be the centralised point of control for our style. We should have our ideals in another repo and have js-sdk, react-sdk, this and web depend on and relax that.

File: package.json (hunk 1 above)
Comment (JorikSchellekens) [reply package.json:32]: I'll ask around

**Comment by t3chguy** [comment]
> The js-sdk's eslint shouldn't be the centralised point of control for our style. We should have our ideals in another repo and have js-sdk, react-sdk, this and web depend on and relax that.

I don't think this is controversial and it has been mentioned in the past. I think now is a better time than any to do so

**Comment by JorikSchellekens** [comment]
Alright, we'll switch to the separate eslint package in a different pr.

**Comment by JorikSchellekens** [comment]
Thanks M

File: package.json (hunk 1 above)
Comment (JorikSchellekens) [reply package.json:32]: Seems like noone has any opinions on it

**Pull request merged**